pkg/controller/stack/stack_controller.go

```diff
@@ -781,48 +785,91 @@ func (sess *reconcileStackSession) DestroyStack() error {
 }
 
 // SetupGitAuth sets up the authentication option to use for the git source
-// repository of the stack.
-func (sess *reconcileStackSession) SetupGitAuth(namespace string) (*auto.GitAuth, error) {
-	var err error
-	namespacedName := types.NamespacedName{Name: sess.stack.GitAuthSecret, Namespace: namespace}
+// repository of the stack. If neither gitAuth or gitAuthSecret are set,
```

Comment:
I will try to get some unit test coverage here — its getting a bit onerous to do everything through the E2E test.

Reply:
FWIW I did test this manually.

Reply:
Updated with tests.